pkg/commands/standard/raw.go

```diff
@@ -0,0 +1,241 @@
+		log.Fatal(err)
+	}
+	defer res.Body.Close()
+	io.Copy(out, res.Body)
+	return res, err
+}
+
+// getBodyFromFlags returns a ReadCloser that represents the body regardless of the parameters used
+func getBodyFromFlags(bodyFile string, bodyString string, bodyParams map[string]string) (body io.ReadCloser, mimeType string, error error) {
+	if multipleBodyParamsSet(bodyString, bodyFile, bodyParams) {
+		return nil, "", errors.New("Only one of body-path, body-string, and body-params can be used at one time")
+	}
+	if bodyString != "" {
+		mimeType := detectStringMimeType(bodyString)
+		return ioutil.NopCloser(strings.NewReader(bodyString)), mimeType, nil
+	}
+	if bodyFile != "" {
+		mimeType := detectFileMimeType(bodyFile)
+		file, err := os.Open(bodyFile)
+		return file, mimeType, err
+	}
+	if bodyParams != nil && len(bodyParams) > 0 {
+		body, err := buildBodyFromParams(bodyParams)
+		return body, "application/json", err
+	}
+	return ioutil.NopCloser(strings.NewReader("")), "", nil
+}
+
+func buildBodyFromParams(params map[string]string) (io.ReadCloser, error) {
+	rootJSonNode := make(map[string]interface{})
+	for key_, value := range params {
+		keyNameAndType := strings.Split(key_, ":")
+		key := keyNameAndType[0]
+		valueType := "string"
+		if len(keyNameAndType) > 2 {
+			return nil, errors.New("invalid key format: " + key_)
+		}
+		if len(keyNameAndType) == 2 {
+			valueType = keyNameAndType[1]
+		}
+		keyParts := strings.Split(key, ".")
+		for _, part := range keyParts {
+			if part == "" {
+				return nil, errors.New("invalid key format: " + key_)
+			}
+		}
+		jsonNode := rootJSonNode
+		for i := 0; i < len(keyParts)-1; i++ {
+			keyPart := keyParts[i]
+			if jsonNode[keyPart] == nil {
+				newNode := make(map[string]interface{})
+				jsonNode[keyPart] = newNode
+				jsonNode = newNode
+			}
+		}
+		lastKey := keyParts[len(keyParts)-1]
+		switch valueType {
+		case "int", "integer", "number":
+			intValue, intErr := strconv.Atoi(value)
+			if intErr != nil {
+				floatValue, floatErr := strconv.ParseFloat(value, 64)
+				if floatErr != nil {
+					if floatErr != nil {
+						return nil, floatErr
+					}
+				}
+				jsonNode[lastKey] = floatValue
+			}
+			jsonNode[lastKey] = intValue
+
+		case "bool", "boolean":
+			boolValue, err := strconv.ParseBool(value)
+			if err != nil {
+				return nil, err
+			}
+			jsonNode[lastKey] = boolValue
+		case "string":
+			jsonNode[lastKey] = value
+		default:
+			return nil, errors.New("invalid key format: " + key_)
+		}
+
+	}
+	jsonBytes, err := json.Marshal(rootJSonNode)
+	if err != nil {
+		return nil, err
+	}
+	return ioutil.NopCloser(bytes.NewReader(jsonBytes)), nil
+}
+
+//detectFileMimeType detects the content-type of the file by checking against various formats
+func detectFileMimeType(filePath string) string {
+	isJsonFile := strings.HasSuffix(strings.ToLower(filePath), ".json")
+	if isJsonFile {
+		return applicationJson
+	}
+	isQvfFile := strings.HasSuffix(strings.ToLower(filePath), ".qvf")
```

Comment:
We should probably also consider adding `.qvw` and `qvd` here.

Reply:
We can't upload qvd's I think, since the datafiles API isn't public, and use of it is discouraged, but qvw definitely.

Reply:
Added  qvf, qvw, qvd and qvx. Note this is only a fast route so if we miss a type it will still work but take slightly more memory and time to determine the mime type